CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/SpellChecker.java

```diff
@@ -0,0 +1,196 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.text;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.input.MouseEvent;
+import javafx.stage.Popup;
+import org.apache.commons.lang3.StringUtils;
+import org.languagetool.JLanguageTool;
+import org.languagetool.MultiThreadedJLanguageTool;
+import org.languagetool.language.AustralianEnglish;
+import org.languagetool.rules.Rule;
+import org.languagetool.rules.RuleMatch;
+/**
+ * Handles the SpellChecking functions of SpellCheckingTextArea. SpellChecker
+ * evaluates incorrect words/phrases and pops up the suggestions when the user
+ * prompts
+ *
+ * @author Auriga2
+ */
+public final class SpellChecker {
+
+    private final SpellCheckingTextArea textArea;
+    private static final List<String> misspells = new ArrayList<>();
+    private List<RuleMatch> matches = new ArrayList<>();
+    private int indexOfMisspelledTextUnderCursor;       // position of the current misspelled text in misspells list
+    private final ListView<String> suggestions = new ListView<>(FXCollections.observableArrayList());
+    private final JLanguageTool langTool;
+    private Popup popup;
+    private boolean turnOffSpellChecking = false;
+    private int startOfMisspelledTextUnderCursor;
+    private int endOfMisspelledTextUnderCursor;
+    private static final Logger LOGGER = Logger.getLogger(SpellChecker.class.getName());
+
+    public SpellChecker(final SpellCheckingTextArea spellCheckingTextArea) {
+        textArea = spellCheckingTextArea;
+        langTool = new MultiThreadedJLanguageTool(new AustralianEnglish()); //Can pass in the language when supporting multiple languages
+
+        initialize();
+    }
+
+    /**
+     * set a listener on the list views selection model to get the text of the
+     * selected row.
+     */
+    private void initialize() {
+        for (Rule rule : langTool.getAllRules()) {
+            if (rule.getId().equals("UPPERCASE_SENTENCE_START")) {
+                langTool.disableRule(rule.getId());
+            }
+        }
+
+        //initialize langtool to prevent the spell checking being too slow at the first word after loading costy
+        try {
+            matches = langTool.check("random text");
+        } catch (final IOException ex) {
+            LOGGER.log(Level.SEVERE, ex.getLocalizedMessage());
+        }
+
+        //initialize popup
+        popup = new Popup();
```

Comment:
perhaps the rest of the code after the catch should be moved into the try block given we're logging an error in the catch at which point we probably don't want to execute anything further in the function